servicetalk-http-api/src/main/java/io/servicetalk/http/api/BlockingStreamingToStreamingService.java

```diff
@@ -65,9 +65,7 @@
     }
 
     private static HttpExecutionStrategy serviceInvocationStrategy(final HttpExecutionStrategyInfluencer influencer) {
-        HttpExecutionStrategy httpExecutionStrategy = influencer.influenceStrategy(DEFAULT_STRATEGY);
-        assert httpExecutionStrategy.isMetadataReceiveOffloaded() : "This will deadlock!";
-        return httpExecutionStrategy;
+        return influencer.influenceStrategy(DEFAULT_STRATEGY);
```

Comment:
Previously, we used this conversion service as part of the computation of the main strategy for the builder. Therefore, we tried to prevent the case when users could misconfigure execution strategy for the blocking service.
For gRPC, we use this service between the router and the final RPC method and we may already be on executor thread.